### PR TITLE
MTM-48361 Updated Sring Boot to 2.7.6

### PR DESCRIPTION
**Pull request by kaczynskid-sag**
This update forced some significant changes in microservice security configuration:
 - There was a cycle: `EnableWebSecurityConfiguration` depending on `CumulocityOAuthMicroserviceFilter`, depending on `AuthenticationManager` that was configured by `EnableWebSecurityConfiguration`. This was resolved using `AuthenticationConfiguration`, provided by `@EnableGlobalAuthentication` and used under the hood by `@EnableWebSecurity`, to deliver a lazy reference to `AuthenticationManager`. Future improvements in OAuth related configurations should re-evaluate it's dependency on `AuthenticationManager` and strive to make it more straightforward.
 - Deprecated `WebSecurityConfigurerAdapter` is no longer used and replaced by explicit `SecurityFilterChain` definition. This means that clients will no longer be allowed to use the adapter, as Spring Boot forces one or the other and will have to migrate when updating to this version.
 - Resorted observability related endpoints exclusions from `SecurityFilterChain` and converted them into independent `WebSecurityCustomizer`.
 - `@ComponentScan` was removed and replaced by explicit `@Configuration` classes:
   - `CumulocityOAuthConfiguration` 
   - `PrePostFiltersConfiguration`

Other than that there we changes required to adapt to new version of `SpringApplicationRunListener` interface and some constants were moved around inside Spring Boot code.

Also `rest-assured` needed to be updated to be compatible with this version of Spring Boot.

Maintenance versions of a few other dependencies were also updated to match the new Spring Boot version.

### CODE REVIEW

File: .jenkins/github/pre-merge.jenkinsfile [hunk 1]
```diff
@@ -183,7 +183,7 @@ def build_agents() {
         dir('cumulocity-agents') {
           stage('agents build & verify') {
             try {
-              sh '.jenkins/scripts/mvn.sh package'
+              sh ".jenkins/scripts/mvn.sh package -Dcumulocity.core.version=${VERSIONS.release}"
```
Comment (kaczynskid-sag) [inline .jenkins/github/pre-merge.jenkinsfile:186]: Unrelated fix to use up to date `3rd-license-maven-plugin`.

File: microservice/security/src/main/java/com/cumulocity/microservice/security/token/OAuthPostAuthorizationContextProvider.java [hunk 2]
```diff
@@ -6,28 +6,22 @@
...
-import org.springframework.stereotype.Component;
 import org.springframework.util.StringUtils;
 
 @Slf4j
-@Component
 class OAuthPostAuthorizationContextProvider implements PostAuthorizationContextProvider<SecurityContext> {
 
     private final String applicationName;
 
     //Optional dependency
```
Comment (c8y-kh4mell) [inline microservice/security/src/main/java/com/cumulocity/microservice/security/token/OAuthPostAuthorizationContextProvider.java:18]: seems not optional anymore (line 30)

**Review by c8y-kh4mell — DISMISSED** [review]

File: microservice/pom.xml [hunk 3]
```diff
@@ -57,6 +58,13 @@
                 <version>${jetty.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>io.rest-assured</groupId>
+                <artifactId>rest-assured-bom</artifactId>
+                <version>${rest-assured.version}</version>
+                <type>pom</type>
```
Comment (ghost) [inline microservice/pom.xml:65]: `<scope>import</scope>`?

File: microservice/security/src/main/java/com/cumulocity/microservice/security/token/CumulocityOAuthConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        private AuthenticationConfiguration authenticationConfiguration;
+        @Autowired(required = false)
+        private AuthenticationEntryPoint authenticationEntryPoint;
+        @Autowired
+        private ContextService<UserCredentials> userContextService;
+
+        @Bean
+        public CumulocityOAuthMicroserviceFilter cumulocityOAuthMicroserviceFilter()
+                throws Exception {
+            return new CumulocityOAuthMicroserviceFilter(authenticationConfiguration.getAuthenticationManager(),
```
Comment (ghost) [inline microservice/security/src/main/java/com/cumulocity/microservice/security/token/CumulocityOAuthConfiguration.java:69]: Can we use bean reference `authenticationConfiguration.getAuthenticationManager()` when `proxyBeanMethods = false`?

File: microservice/security/src/main/java/com/cumulocity/microservice/security/token/OAuthPostAuthorizationContextProvider.java (hunk 2 above)
Comment (ghost) [reply microservice/security/src/main/java/com/cumulocity/microservice/security/token/OAuthPostAuthorizationContextProvider.java:18]: It is still optional, see
```
OAuthPostAuthorizationContextProvider provider = new OAuthPostAuthorizationContextProvider(applicationName);
        subscriptionsService.ifAvailable(provider::setSubscriptionsService);
        return provider;
```

File: microservice/security/src/main/java/com/cumulocity/microservice/security/token/CumulocityOAuthConfiguration.java (hunk 4 above)
Comment (kaczynskid-sag) [reply microservice/security/src/main/java/com/cumulocity/microservice/security/token/CumulocityOAuthConfiguration.java:69]: `org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration#getAuthenticationManager()` is _not_ a `@Bean` method. As mentioned in PR description it's the way it's done in Spring under the hood to facilitate global configuration using `AuthenticationManagerBuilder`.

File: microservice/pom.xml (hunk 3 above)
Comment (kaczynskid-sag) [reply microservice/pom.xml:65]: added

**Review by ghost — APPROVED** [review]

File: microservice/security/src/main/java/com/cumulocity/microservice/security/filter/provider/HttpContextProvider.java [hunk 5]
```diff
@@ -1,19 +1,21 @@
...
 import javax.servlet.http.HttpServletRequest;
 import java.util.Arrays;
 import java.util.Optional;
 
-import static com.cumulocity.microservice.security.filter.util.HttpRequestUtils.*;
+import static com.cumulocity.microservice.security.filter.util.HttpRequestUtils.TFA_TOKEN_HEADER;
+import static com.cumulocity.microservice.security.filter.util.HttpRequestUtils.XSRF_TOKEN_HEADER;
+import static com.cumulocity.microservice.security.filter.util.HttpRequestUtils.X_CUMULOCITY_APPLICATION_KEY;
+import static com.cumulocity.microservice.security.filter.util.HttpRequestUtils.authorizationHeader;
+import static com.cumulocity.microservice.security.filter.util.HttpRequestUtils.hasAuthorizationHeader;
```
Comment (jkot-c8y) [inline microservice/security/src/main/java/com/cumulocity/microservice/security/filter/provider/HttpContextProvider.java:16]: minor: not needed imports refactor

File: microservice/security/src/main/java/com/cumulocity/microservice/security/filter/provider/SpringSecurityContextProvider.java [hunk 6]
```diff
@@ -5,28 +5,22 @@
...
 import org.springframework.security.core.context.SecurityContext;
 import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.stereotype.Component;
 import org.springframework.util.StringUtils;
 
+import java.util.Optional;
+
 @Slf4j
-@Component
+@RequiredArgsConstructor
```
Comment (jkot-c8y) [inline microservice/security/src/main/java/com/cumulocity/microservice/security/filter/provider/SpringSecurityContextProvider.java:18]: minor: maybe declare constructor directly (without Lombok), as you do in the other place (`CumulocityOAuthMicroserviceFilter`)

**Review by jkot-c8y — APPROVED** [review]
